components/ContractViewer/ContractTreeNode.tsx

```diff
@@ -0,0 +1,661 @@
+          to: address,
+          data,
+          value: funcData.value,
+        }
+
+        return walletClient.sendTransaction(props as any)
+          .then((res: any) => {
+            let decoded = decodeFunctionResult({
+              abi: [funcAbi],
+              data: res.data,
+            })
+
+            if (funcAbi.outputs.length == 1) {
+              decoded = [decoded]
+            }
+
+            return decoded.map((val: any) => val.toString())
+          })
+      })
+      .catch((err: any) => {
+        setStatus(err.toString())
+      })
+  }
+
+  const setCallStatus = () => {
+    const [data, error] = encodeCalldata()
+    if (error) {
+      setStatus(error.toString())
+      updateFuncData({ outputs: [] })
+      return
+    }
+
+    ethCall(data)
+      .then((decoded: any) => {
+        setStatus('')
+        updateFuncData({ outputs: decoded })
+      })
+      .catch((err: any) => {
+        setStatus(err.toString())
+        updateFuncData({ outputs: [] })
+      })
+  }
+
+  const setEncodeStatus = () => {
+    const [data, error] = encodeCalldata()
+    setStatus('calldata: ' + (data || error))
+  }
+
+  return (
+    <TreeItemBasic
+      nodeId={`function_${id}_${funcAbi.name}`}
+      title={funcName}
+      subtitle={subtitle}
+    >
+      <div className="flex flex-col gap-2 text-black-500 my-2 mr-4">
+        <span className="text-xs dark:text-gray-200">
+          4byte selector: {toFunctionSelector(funcAbi)}
+        </span>
+        {funcAbi && <ParamsBox abi={funcAbi} reducer={reducer} />}
+        {(funcAbi.inputs.length > 0 || funcAbi.stateMutability == 'payable') && (
+          <div className="flex gap-1">
+            <Button onClick={setCallStatus} variant="contained">
+              {funcAbi?.outputs?.length > 0
+                ? 'Call'
+                : 'Call (no ret)'}
+            </Button>
+            <Button onClick={setEncodeStatus} variant="contained">
+              Encode
+            </Button>
+            <Button onClick={ethSendTransaction} variant="contained" color="secondary">
+              Send
+            </Button>
+          </div>
+        )}
+        {(funcAbi && funcAbi?.outputs?.length > 0) && <ReturnDataBox abi={funcAbi} reducer={reducer} />}
+        {status && (
+          <input
+            type="button"
+            className="cursor-pointer hover:bg-blue-100 dark:hover:bg-blue-900 text-left whitespace-pre-line break-all text-sm dark:text-gray-100"
+            onClick={() =>
+              navigator.clipboard.writeText(status)
+            }
+            value={'📋 copy'}
+          />
+        )}
+        <p className="text-xs text-gray-500 break-words">
+          {spaceBetween(status || '')}
+        </p>
+      </div>
+    </TreeItemBasic>
+  )
+}
+
+type StorageLayoutItemProps = {
+  id: string
+  address: string
+  storage: {
+    astId: number
+    label: string
+    contract: string
+    offset: number
+    slot: any // string | number
```

Comment:
If it can just be string or number, why have it as a `any` ? 🤔

Reply:
the other `any`s did not bother you?

Reply:
Ideally we don't want `any` if we can have a specific type.

This part of what I was refereeing to by us having typechecking.

Reply:
removed the type comment